front-end/src/main/kotlin/org/jetbrains/kotlin/spec/loader/GithubTestsLoader.kt

```diff
@@ -0,0 +1,139 @@
+import js.externals.jquery.JQueryPromise
+import js.externals.jquery.JQueryXHR
+import js.externals.jquery.`$`
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonConfiguration
+import org.jetbrains.kotlin.spec.entity.Section
+import org.jetbrains.kotlin.spec.entity.SectionTestMap
+import org.jetbrains.kotlin.spec.entity.test.Test
+import org.jetbrains.kotlin.spec.entity.test.TestPlace
+import org.jetbrains.kotlin.spec.entity.test.parameters.TestInfo
+import org.jetbrains.kotlin.spec.entity.test.parameters.testArea.TestArea
+import org.jetbrains.kotlin.spec.utils.format
+import kotlin.browser.window
+import kotlin.js.Promise
+
+interface GithubTestsLoader {
+    enum class TestFileType { SPEC_TEST, IMPLEMENTATION_TEST }
+
+    companion object {
+        private const val RAW_GITHUB_URL = "https://raw.githubusercontent.com/JetBrains/kotlin"
+
+        private const val SPEC_TEST_DATA_PATH = "compiler/tests-spec/testData"
+        private const val LINKED_SPEC_TESTS_FOLDER = "linked"
+
+        const val DEFAULT_BRANCH = "spec-tests"
+
+        fun getBranch() = window.localStorage.getItem("spec-tests-branch") ?: DEFAULT_BRANCH
+
+        fun loadTestFileFromRawGithub(
+                path: String,
+                testPathInSpec: String? = null,
+                customFolder: String? = null,
+                testInfo: TestInfo,
+                testPlace: TestPlace,
+                testArea: TestArea
+        ): Promise<Test> {
+            return Promise { requestResolve, requestReject ->
+                val testContainerMap = mutableMapOf<String, String>()
+
+                val testsPaths = TestsPaths(customFolder, path, testPathInSpec) //todo get rid of ?
+
+                val queryForTestArea: JQueryPromise<Unit> = getQueryForTest(testContainerMap, requestReject, testsPaths)
+
+                `$`.`when`(queryForTestArea).then({ _: Any?, _: Any ->
+                    requestResolve(Test(testInfo, testPlace, testContainerMap["content"]!!, testArea))
+                })
+            }
+        }
+
+        fun loadTestMapFileFromRawGithub(
+                path: String,
+                testPathInSpec: String? = null,
+                testType: TestFileType,
+                customFolder: String? = null,
+                testAreasToLoad: Array<TestArea>
+        ): Promise<MutableMap<TestArea, SectionTestMap>> {
+            return Promise { resolve, _ ->
+                val resultMap = mutableMapOf<TestArea, SectionTestMap>()
+                val testsPaths = TestsPaths(customFolder, path, testPathInSpec)
+                val querySet = mutableSetOf<JQueryPromise<Unit>>()
+                testAreasToLoad.forEach { testArea ->
+                    querySet.add(
+                            getQueryForTestMap(testArea, resultMap, testType, testsPaths)
+                    )
+                }
+                `$`.`when`(*(querySet.toTypedArray()))
+                        .then({ _: Any?, _: Any ->
```

Comment:
It seems `done` with one lambda or something similar can be used instead of then